lib/jsdom/living/nodes/Element-impl.js

```diff
@@ -430,6 +430,58 @@ class ElementImpl extends NodeImpl {
 
     return attr;
   }
+
+  insertAdjacentHTML(position, text) {
+    position = position.toLowerCase();
+
+    let context;
+
+    switch (position) {
+      case "beforebegin":
+      case "afterend":
+        context = this.parentNode;
+
+        if (context === null || context.nodeType === NODE_TYPE.DOCUMENT_NODE) {
+          throw new DOMException(DOMException.NO_MODIFICATION_ALLOWED_ERR, "Modifications are not allowed for this" +
```

Comment:
I don't believe the existing test suite covers these exceptions, so adding to-upstream tests for them would be lovely.